src/app/components/kudos-user-history/kudos-user-history.component.js

```diff
@@ -0,0 +1,229 @@
+                else if (self.receivedOperations) updateListReceived(self.thisUsersEmail, false);
+                else if (self.gaveOperations) updateListGave(self.thisUsersEmail, false);
+                else if (self.challengeOperations) updateListChallenges(self.thisUsersEmail, false);
+            }
+
+            if (val == 'less') {
+                self.transactionEndingIndex = 6;
+                self.showMoreButton = true;
+                self.showLessButton = false;
+            }
+        }
+        
+        function showList(email, trigger) {
+            self.transactionEndingIndex = 6;
+
+            if (self.allOperations) updateListAll(email, trigger);
+            else if (self.receivedOperations) updateListReceived(email, trigger);
+            else if (self.gaveOperations) updateListGave(email, trigger);
+            else if (self.challengeOperations) updateListChallenges(email, trigger);
+        }
+
+        function changeRadioValue(value) {
+            if (self.historyRadioBox != value) {
+                self.historyRadioBox = value;
+            }
+        }
+
+        function changeView(value) {
+            if (value == 'all') {
+                self.receivedOperations = false;
+                self.allOperations = true;
+                self.gaveOperations = false;
+                self.challengeOperations = false;
+            } else if (value == 'received'){
+                self.receivedOperations=true;
+                self.allOperations=false;
+                self.gaveOperations=false;
+                self.challengeOperations=false;
+            } else if (value == 'gave'){
+                self.receivedOperations=false;
+                self.allOperations=false;
+                self.gaveOperations=true;
+                self.challengeOperations=false;
+            } else if (value == 'challenges'){
+                self.receivedOperations=false;
+                self.allOperations=false;
+                self.gaveOperations=false;
+                self.challengeOperations=true;
+            }
+
+            self.showLessButton = false;
+            self.transactionEndingIndex = 6;
+            self.historyRadioBox = 'wonLost';
+        }
+
+        function getTemplate(page) {
+            if (this.page == true){
+                return 'app/components/kudos-user-history/kudos-user-history-page.html';
+            } else {
+                return 'app/components/kudos-user-history/kudos-user-history.html';
+            }
+        }
+
+        function spin() {
+            $timeout(function () {
+                self.spin = false;
+            }, 1000);
+        }
+
+    };
+
+    UserHistoryController.$inject = ['Resources', 'UserHistoryService', '$timeout'];
+
+    angular.module('myApp.components.userHistory', [])
+        .component('kudosUserHistory', {
+            template: '<ng-include src="self.getTemplate()"/>' ,
+            bindings: {
+                user: '=',
```

Comment:
Are you updating `self.user` anywhere? Do you need this two-way-binding? One way bindings were introduced to components, since usually components only need read-access to any data that is provided into their isolate scope and thus, there is no need to watch for the changes on the variables on the parent controller. This makes it faster and less prone to errors.